grammar.js

```diff
@@ -39,72 +43,111 @@ module.exports = grammar({
+    list_shape_statement: $ => seq('list', $.identifier, optional($.mixins), $.shape_members),
+    set_shape_statement: $ => seq('set', $.identifier, optional($.mixins), $.shape_members),
+    map_shape_statement: $ => seq('map', $.identifier, optional($.mixins), $.shape_members),
+    structure_shape_statement: $ => seq(
+      'structure',
+      $.identifier,
+      optional($.structure_resource),
+      optional($.mixins),
+      $.shape_members
+    ),
+    union_shape_statement: $ => seq('union', $.identifier, optional($.mixins), $.shape_members),
 
-    service_shape_statement: $ => seq('service', $.identifier, $.node_object),
-    operation_shape_statement: $ => seq('operation', $.identifier, $.node_object),
-    resource_shape_statement: $ => seq('resource', $.identifier, $.node_object),
+    service_shape_statement: $ => seq('service', $.identifier, optional($.mixins), $.node_object),
+    operation_shape_statement: $ => seq('operation', $.identifier, optional($.mixins), $.operation_body),
+    resource_shape_statement: $ => seq('resource', $.identifier, optional($.mixins), $.node_object),
 
+    enum_shape_statement: $ => seq(
+      choice('enum', 'intEnum'),
+      $.identifier,
+      optional($.mixins),
+      $.enum_members
+    ),
+    enum_members: $ => seq('{', repeat($.enum_member), '}'),
+    enum_member: $ => seq($.identifier, optional($.value_assignment)),
 
-    shape_members: $ => choice($.empty_shape_members, $.populated_shape_members),
-    empty_shape_members: $ => seq('{', '}'),
-    populated_shape_members: $ => seq(
+    shape_members: $ => seq(
       '{',
-      $.shape_member_kvp,
-      repeat(seq(',', $.shape_member_kvp)),
-      optional(','),
+      repeat($.shape_member),
       '}'
     ),
-    shape_member_kvp: $ => choice(
-      seq($.trait_statements, seq($.identifier, ':', $.shape_id)),
-      seq($.identifier, ':', $.shape_id)
+    shape_member: $ => seq(
+      optional($.trait_statements),
+      choice(
+        $.shape_member_kvp,
+        $.shape_member_elided
+      ),
+      optional($.value_assignment)
+    ),
+    shape_member_kvp: $ => seq($.identifier, ':', $.shape_id),
+    shape_member_elided: $ => seq('$', $.identifier),
+
+    operation_body: $ => seq('{', repeat($.operation_member), '}'),
+    operation_member: $ => choice(
+      $.shape_member_kvp,
+      seq($.identifier, $.inline_structure),
+      $.operation_errors
+    ),
+    operation_errors: $ => seq($.identifier, ':', '[', repeat($.operation_error), ']'),
+    operation_error: $ => $.identifier,
+
+    inline_structure: $ => seq(
+      ':=',
+      optional($.trait_statements),
+      optional($.structure_resource),
+      optional($.mixins),
+      $.shape_members
     ),
 
     trait_statements: $ => repeat1($.trait),
-    trait: $ => seq('@', $.shape_id, repeat($.trait_body)),
-    trait_body: $ => seq('(', $.trait_body_value, ')'),
+    trait: $ => seq('@', $.shape_id, optional($.trait_body)),
```

Comment:
The use of `repeat` looks to be a mistake here and should be `optional` instead. See https://awslabs.github.io/smithy/2.0/spec/idl.html#grammar-token-smithy-Trait